scripts/cluster.sh

```diff
@@ -215,6 +215,329 @@ function download_iso() {
     log "INFO" "ISO downloaded successfully to ${ISO_FOLDER}"
 }
 
+function get_iso() {
+    # Unified function that handles ISOs for both master and worker nodes
+    # Parameters:
+    #   $1 - cluster_name: Name of the cluster
+    #   $2 - node_type: "master" or "worker"
+    #   $3 - action: "url" or "download"
+    #   $4 - download_path: Path to download ISO (if action is "download")
+    #   $5 - iso_type: "minimal" or "full", defaults to "minimal"
+    
+    local cluster_name="$1"
+    local node_type="$2"
+    local action="$3"
+    local download_path="$4"
+    local iso_type="${5:-minimal}"  # Default to minimal if not specified
+    local day2_cluster_name="${cluster_name}-day2"
+    
+    # Export for use in other functions
+    export CLUSTER_NAME="${cluster_name}"
+    export ISO_FOLDER="${download_path:-${ISO_FOLDER}}"
+    export ISO_TYPE="${iso_type}"
+    
+    log "INFO" "ISO request: cluster=${cluster_name}, type=${node_type}, action=${action}, path=${download_path}, iso_type=${iso_type}"
+    
+    if [[ "${node_type}" != "master" && "${node_type}" != "worker" ]]; then
```

Comment:
Node type should not matter, just download iso from cluster name, you can provide day2 boolean probably and no need for this check